torch/csrc/autograd/profiler_legacy.cpp

```diff
@@ -230,7 +230,7 @@ void ProfilerThreadLocalState::pushRange(
       evt.setExtraArgs(saveExtraArgs(fn));
       evt.setFlops(computeFlops(std::string(fn.name().str()), evt.extraArgs()));
     }
-#ifndef C10_MOBILE
+#if !defined DBUILD_LITE_INTERPRETER && !defined C10_MOBILE
```

Comment:
C10_BUILD is wired with aten. It's probably better to remove the `DBUILD_LITE_INTERPRETER` when getting lite-interpreter decoupled from jit at aten level.

Reply:
C10_MOBILE looks like internal only. Could you set it in CMake too? Something like:
```
if (BUILD_LITE_INTERPRETER)
  CMAKE_ARGS+=("-DBUILD_LITE_INTERPRETER=ON")
```
I have a feeling that if we enable C10_MOBILE, the binary size of lite interpreter could be reduced significantly.

Reply:
Do you mean this?
````
if (BUILD_LITE_INTERPRETER)
  CMAKE_ARGS+=("-DC10_MOBILE =ON")
```
? I tried to enable `C10_MOBILE` manually in cmake file and build it with `BUILD_LITE_INTERPRETER=ON`. There are lots of linker error, most of them look like aten.

Reply:
Yes I meant to set C10_MOBILE. Thanks for the correction.

Reply:
Please add a comment into the code, to explain why we have an additional macro, and the plan to unify it to C10_MOBILE in near future. In general the macro usage should be limited to make the code cleaner and less error prone.

Reply:
Address the comment in https://github.com/pytorch/pytorch/pull/52573.